- switching to a toolset rather than simply retrival

gambol99 opened the pull request:
summary: addresses https://github.com/UKHomeOffice/s3secrets/issues/8 and should have jumped straight to this rather than the last PR, anywho! .. Essentially, you can now use it a toolset to secrets and kms retrieving, listing, upload, catting and removing along with some kms subcommands for encryption/decryption and lists
- add the secrets and kms commands
- we can now, list, get, cat and delete secrets from the command line
- we can list, encrypt and decrypt via the kms from the command line
- added listing coloring (pretty pretty)
- fixed up the filtering of the files
- applying the gofmt, lint and vet
- adding the listing of the s3 buckets

gambol99 committed:
- switching to a toolset rather than simply retrival
- add the secrets and kms commands
- we can now, list, get, cat and delete secrets from the command line
- we can list, encrypt and decrypt via the kms from the command line
- added listing coloring
- fixed up the filtering of the files
- applying the gofmt, lint and vet
- adding the listing of the s3 buckets

vaijab commented:
Quite a big change to the tool, but looks nice and I like most of the new features. Well done!

Few comments, but otherwise looks good, but before you merge, please update the `README.md`. :-)

Thanks!

gambol99 commented:
cool ... changed the env variable to AWS_DEFAULT_REGION, removing the default region (now you have to -r or export the var), long description examples changed and fixed the typo :-)

vaijab commented:
> now you have to -r or export the var

I am not sure how go aws library works (being too lazy to check), but for example boto does not check for `region =` in credentials file, so in stacks` I check for that separately so that you don't have to specify a region as long as stacks can figure out what profile you're using.

AWS cli / api libs configuration is a bit weird, like everything else in AWS :-), they are two configuration files, `.aws/config` and `.aws/credentials`. So you put region, etc into one file and credentials into another file. Also profile definition syntax is different depending on which file you use. Crazy :-)
